source/common/config/legacy_grpc_subscription_impl.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/config/grpc_mux.h"
+#include "envoy/config/subscription.h"
+#include "envoy/event/dispatcher.h"
+
+#include "common/common/logger.h"
+
+#include "xds/core/v3/resource_locator.pb.h"
+
+namespace Envoy {
+namespace Config {
+
+/**
+ * Adapter from typed Subscription to untyped GrpcMux. Also handles per-xDS API stats/logging.
+ */
+class LegacyGrpcSubscriptionImpl : public Subscription,
```

Comment:
How come we need a legacy subscription? The subscription is largely independent of the gRPC mux itself, it would seem not a lot of logic should change here.

Reply:
Largely due to `init_fetch_timeout` timer moving to watches. I think having a dedicated implementation is easier/safer/more maintainable than having conditionals in one common implementation.